cpp/src/arrow/util/test_common.h

```diff
@@ -85,4 +88,18 @@ inline void AssertIteratorExhausted(Iterator<T>& it) {
 
 Transformer<TestInt, TestStr> MakeFilter(std::function<bool(TestInt&)> filter);
 
+class MockExecutor : public internal::Executor {
```

Comment:
Ideally, we should start categorizing our test utilities better instead of dumping them all in generic header files.

Reply:
There is a bit of categorization based on folder (e.g. `util/test_common.h` vs `io/test_common.h`).  This file isn't really all that large.  However, it could probably be split into `test_iterator_common.h` and `test_thread_pool_common.h`.  Or am I missing the point?

Reply:
I think we have `arrow/testing/future_util.h`. We could move the mock executor into that file, or into a separate `arrow/testing/executor_util.h`.

Reply:
I went ahead and added `executor_util.h`.